Docs: Add documentation for inspector in Explore (#33408)
grafana/grafana: docs/sources/explore/explore-inspector.md

+++
title = "Inspector in Explore"
weight = 400
+++

# Inspector in Explore

> The inspector helps you understand and troubleshoot your queries. You can inspect the raw data, export that data to a comma-separated values (CSV) file, export log results in TXT format, and view query requests.

Author: I wouldn't call it query inspector, just inspector as it inspects more than just the query - data, data structure, ... .

## Inspector UI

> The inspector consists of four tabs:

Reviewer: ```suggestion
The inspector has four tabs:
```

- **Data tab -** Shows the raw data returned by the query.
- **Stats tab -** Shows how long your query takes and how much it returns.
- **JSON tab -** Allows you to view and copy the data JSON and data frame structure JSON.

> - **Query tab -** Shows you the requests to the server sent when Grafana queries the data source.

Reviewer: List them as they show in the UI - Stats, Query inspector, JSON, and Data

Author: Fixed in bd89ecf. I have also added Error tab as https://github.com/grafana/grafana/pull/33412 adds it.

## Inspector tasks

> Tasks you can perform in the inspector are described below.

Reviewer: ```suggestion
You can perform a variety of tasks in the Explore inspector.
```

### Open the Inspector

> 1. In Grafana, navigate to the Explore and run the query you would like to inspect.

Author: ```suggestion
1. Run the query you would like to inspect.
```

1. Click the **Inspector** button.

The inspector pane opens on the bottom of the screen.

### Inspect raw query results

View raw query results in a table. This is the data returned by the query.

1. Open the inspector and then click the **Data** tab.

> 1. If you have run multiple queries or queries multiple nodes, then you have additional options.

Reviewer: ```suggestion
For multiple queries or for queries multiple nodes, there are additional options.
```

> 1. If you have run multiple queries or queries multiple nodes, then you have additional options.

Author: Fixed in bd89ecf

> - **Show data frame -** Choose which result set data you want to view.

Reviewer: ```suggestion
- **Show data frame:** Select the result set data you want to view.
```

>   - **Series joined by time -** View raw data from all your queries at once, one result set per column. Click a column heading to reorder the data.

Reviewer: ```suggestion
- **Series joined by time:** View the raw data from all of your queries at once, one result set per column. You can click a column heading to sort the data.
```

Reviewer: I did not see this option in https://play.grafana.org/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22demo.robustperception.io%22,%7B%22exemplar%22:true,%22expr%22:%22ALERTS%22,%22requestId%22:%22Q-61eb23f3-97e4-4ea2-8247-3eb3bb2ec094-0A%22%7D%5D

Author: It is there: ![image](https://user-images.githubusercontent.com/30407135/116858247-5faabd00-abfe-11eb-8228-c32480330494.png)

### Download raw query results as CSV

Grafana generates a CSV file in your default browser download location. You can open it in the viewer of your choice.

> 1. Open the inspector.

Reviewer: ```suggestion
1. In the **Inspector** tab, raw query results following instructions in (### Inspect raw query results).
```

> 1. Inspect the raw query results as described above. Adjust settings until you see the raw data that you want to export.

Reviewer: ```suggestion
1. Refine query settings until you can see the raw data that you want to export.
```

1. Click **Download CSV**.

> To download a CSV file specifically formatted for Excel, expand the **Data options** and enable the **Download for Excel** toggle before clicking **Download CSV**.

Reviewer: ```suggestion
In order to download a CSV file specifically formatted for Excel, expand **Data options** and then enable the **Download for Excel** toggle before you click the **Download CSV** option.
```

### Download log results as TXT

Grafana generates a TXT file in your default browser download location. You can open it in the viewer of your choice.

1. Open the panel inspector.
1. Inspect the log query results as described above.
1. Click **Download logs**.

### Inspect query performance

The Stats tab displays statistics that tell you how long your query takes, how many queries you send, and the number of rows returned. This information can help you troubleshoot your queries, especially if any of the numbers are unexpectedly high or low.

1. Open the inspector.
1. Navigate to the **Stats** tab.

Statistics are displayed in read-only format.

### View JSON model

Explore and export data and data frame JSON models.

1. Open the inspector and then click the **JSON** tab.
1. In Select source, choose one of the following options:
   - **Data -** Displays a JSON object representing the data that was passed to Explore.
   - **DataFrame structure -** Displays the raw result set.

> 1. You can expand or collapse portions of the JSON to explore it, or you can click **Copy to clipboard** and paste the JSON in another application.

Reviewer: ```suggestion
1. You can expand or collapse portions of the JSON to view separate sections. You can also click the **Copy to clipboard** option to copy JSON body and paste it into another application.
```

### View raw request and response to data source

> 1. Open the panel inspector and then click the **Query** tab.

Author: This is going to be renamed to just query tab in https://github.com/grafana/grafana/pull/33412

+++
title = "Inspector in Explore"
weight = 400
+++

# Inspector in Explore

The inspector helps you understand and troubleshoot your queries. You can inspect the raw data, export that data to a comma-separated values (CSV) file, export log results in TXT format, and view query requests.

## Inspector UI

The inspector has following tabs:

- **Stats tab -** Shows how long your query takes and how much it returns.
- **Query tab -** Shows you the requests to the server sent when Grafana queries the data source.
- **JSON tab -** Allows you to view and copy the data JSON and data frame structure JSON.
- **Data tab -** Shows the raw data returned by the query.
- **Error tab -** Shows the error. Only visible when query returns error.

## Inspector tasks

You can perform a variety of tasks in the Explore inspector.

### Open the Inspector

1. Run the query you would like to inspect.
1. Click the **Inspector** button.

The inspector pane opens on the bottom of the screen.

### Inspect raw query results

You can view raw query results, that is the data returned by the query in a table.

> 1. In the **Inspector** tab, click the **Data** tab.

Reviewer: Since this is just one step, it should be an indented paragraph. Not a numbered step.

Author: ```suggestion
  In the **Inspector** tab, click the **Data** tab.
```

Author: By indented paragraph you mean this (See suggestion above)? @achatterjee-grafana

Reviewer: Perfect. Thanks!
